src/indicators/drought/ecmwf_seasonal/ecmwf_seasonal.py

```diff
@@ -0,0 +1,272 @@
+            "month": [str(x).zfill(2) for x in range(1, 13)]
+            if month is None
+            else str(month).zfill(2),
+            "leadtime_month":[str(x) for x in leadtimes],
+            "area": area.list_for_api(),
+        }
+        logger.debug(f"Query: {query}")
+        return query
+
+    @staticmethod
+    def _read_in_monthly_mean_dataset(
+            filepath_list: List[Path],
+            leadtimes: List[int],
+    ):
+        """
+        Read in the dataset for each date and combine them
+        """
+
+        def _preprocess_monthly_mean_dataset(ds,leadtimes):
+            # step is in timedelta (in nanoseconds), where the timedelta is the end of the valid time of the forecast
+            # since the nanoseconds depends on the length of the month, convert this to the leadtime in months instead to be able to compare across months
+            #other option could be to convert it to the forecasted time, but gets difficult to concat all different publication dates afterwards
+            ds["step"] = leadtimes
+            # ds["step"] = ds["time"] + ds["step"]
+            # time is the publication month of the forecast, add this to the dimensions to be able to merge different times
+            ds = ds.expand_dims("time")
+
+            return ds
+
+        with xr.open_mfdataset(
+                filepath_list, engine="cfgrib", backend_kwargs={"indexpath": ""}, preprocess=lambda d:_preprocess_monthly_mean_dataset(d,leadtimes),
+        ) as ds:
+            return ds
+
+    def _write_to_processed_file(
+        self,
+        country_iso3: str,
+        version: int,
+        ds: xr.Dataset,
+    ) -> Path:
+        filepath = self._get_processed_filepath(
+            country_iso3=country_iso3,
+            version=version,
+        )
+        Path(filepath.parent).mkdir(parents=True, exist_ok=True)
+        logger.info(f"Writing to {filepath}")
+        ds.to_netcdf(filepath)
+        return filepath
+
+    def _get_processed_filepath(
+        self, country_iso3: str, version: int
+    ) -> Path:
+        filename = f"{country_iso3}_{self.cds_name}_v{version}"
+        filename += ".nc"
+        return PROCESSED_DATA_DIR / country_iso3 / ECMWF_SEASONAL_DIR / filename
+
+    def read_processed_dataset(
+        self,
+        country_iso3: str,
+        version: int = DEFAULT_VERSION,
+    ):
+        filepath = self._get_processed_filepath(
+            country_iso3=country_iso3,
+            version=version
+        )
+        return xr.load_dataset(filepath)
+
+class EcmwfSeasonalForecast(EcmwfSeasonal):
+    def __init__(self):
+        super().__init__(
+            year_min=2000,
+            year_max=2021,
+            cds_name="seasonal-monthly-single-levels",
+            dataset=["monthly_mean"],
+            dataset_variable_name="product_type",
+        )
+
+    def download(
+        self,
+        country_iso3: str,
+        area: Area,
+        leadtimes: List[int] = DEFAULT_LEADTIMES,
```

Comment:
Default arguments should never be mutable, because if you change the variable somewhere in the method then the default will also be changed. Best practice is to set it to `None`, and then in the method do:
```python
if leadtimes is None:
    leadtimes = DEFAULT_LEADTIMES
```